src/status_im/contexts/wallet/swap/events.cljs

```diff
@@ -120,16 +116,21 @@
 (rf/reg-event-fx :wallet/swap-proposal-success
  (fn [{:keys [db]} [swap-proposal]]
    (let [last-request-uuid (get-in db [:wallet :ui :swap :last-request-uuid])
+         view-id           (:view-id db)
          request-uuid      (:uuid swap-proposal)
          best-routes       (:best swap-proposal)
          error-response    (:error-response swap-proposal)]
      (when (= request-uuid last-request-uuid)
-       {:db (update-in db
-                       [:wallet :ui :swap]
-                       assoc
-                       :swap-proposal          (first best-routes)
-                       :error-response         (when (empty? best-routes) error-response)
-                       :loading-swap-proposal? false)}))))
+       (cond-> {:db (update-in db
+                               [:wallet :ui :swap]
+                               assoc
+                               :swap-proposal          (first best-routes)
+                               :error-response         (when (empty? best-routes) error-response)
+                               :loading-swap-proposal? false)}
+         (and (empty? best-routes) (= view-id :screen/wallet.swap-set-spending-cap))
+         (assoc :fx [[:dismiss-modal :screen/wallet.swap-set-spending-cap]])
+         (and (empty? best-routes) (= view-id :screen/wallet.swap-confirmation))
+         (assoc :fx [[:navigate-back]]))))))
```

Comment:
Router is unstable and it can return a swap proposal and after auto-refetching it can return an error. Ideally this shouldn't happen, but I am adding this behavior so if the user is in swap confirmation screen or in token approval confirmation screen, we navigate back to setup swap screen so proper error is displayed.

Reply:
Added!